### ISSUE TITLE
Need to normalize

### ISSUE THREAD

**Issue by qzchenwl** [titu1994/neural-image-assessment#4]
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/_impl/keras/applications/imagenet_utils.py#L82

```python
def _preprocess_symbolic_input(x, data_format, mode):
  """Preprocesses a symbolic image tensor.
  Arguments:
      x: symoblic tensor, 3D or 4D.
      data_format: data format of the image tensor.
      mode: One of "caffe", "tf".
          - caffe: will convert the images from RGB to BGR,
              then will zero-center each color channel with
              respect to the ImageNet dataset,
              without scaling.
          - tf: will scale pixels between -1 and 1,
              sample-wise.
  Returns:
      Preprocessed tensor.
  """
  global _IMAGENET_MEAN

  if mode == 'tf':
    x /= 127.5
    x -= 1.
    return x

  if data_format == 'channels_first':
    # 'RGB'->'BGR'
    if K.ndim(x) == 3:
      x = x[::-1, ...]
    else:
      x = x[:, ::-1, ...]
  else:
    # 'RGB'->'BGR'
    x = x[..., ::-1]

  if _IMAGENET_MEAN is None:
    _IMAGENET_MEAN = K.constant(-np.array([103.939, 116.779, 123.68]))
  # Zero-center by mean pixel
  if K.dtype(x) != K.dtype(_IMAGENET_MEAN):
    x = K.bias_add(x, K.cast(_IMAGENET_MEAN, K.dtype(x)), data_format)
  else:
    x = K.bias_add(x, _IMAGENET_MEAN, data_format)
  return x
```
https://github.com/titu1994/neural-image-assessment/blob/master/data_loader.py#L59

1. Which order RGB/BGR does MobileNet accept?
2. Should we use `IMAGENET_MEAN = [103.939, 116.779, 123.68]` or just `[127, 127, 127]`?

**Comment by qzchenwl** [comment]
I see `keras.applications.mobilenet.preprocess_input` in `evaluate_mobilenet`, but not in `data_loader`. Why not `preprocess_input` before training?

**Comment by titu1994** [comment]
MobileNet preprocess_input is in -1 to 1 range. I use that directly for evaluation purposes. 

For the data loaders, the two parse methods that map the file path to an image also have the img - 127.5 / 127.5 line which normalizes to -1 to 1 range.
